src/light.h

```diff
@@ -65,18 +65,17 @@ inline u8 decode_light(u8 light)
 // 0.0 <= return value <= 1.0
 inline float decode_light_f(float light_f)
 {
-	s32 i = (u32)(light_f * LIGHT_MAX + 0.5);
-
-	if (i <= 0)
-		return (float)light_decode_table[0] / 255.0;
-	if (i >= LIGHT_SUN)
-		return (float)light_decode_table[LIGHT_SUN] / 255.0;
-
-	float v1 = (float)light_decode_table[i - 1] / 255.0;
-	float v2 = (float)light_decode_table[i] / 255.0;
-	float f0 = (float)i - 0.5;
-	float f = light_f * LIGHT_MAX - f0;
-	return f * v2 + (1.0 - f) * v1;
+	light_f *= LIGHT_SUN; // should be [0, LIGHT_SUN]
```

Comment:
How about moving this to after line 72 so that the 'return's below happen faster with less calculations?
Most of the day 'light_f' is 1.0 so it's best to immediately return in this situation.

Reply:
```
	if (light_f <= 0.0f)
		return 0.0f;
	if (light_f >= 1.0f)
		return 1.0f;
	light_f *= LIGHT_SUN; // should be [0, LIGHT_SUN]
```